airflow/providers/google/cloud/hooks/cloud_run.py

```diff
@@ -0,0 +1,245 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This module contains a Google Cloud Run Hook."""
+from __future__ import annotations
+
+import json
+import time
+from typing import Any, Callable, Dict, List, Sequence, Union, cast
+
+from google.api_core.client_options import ClientOptions
+from googleapiclient.discovery import build
+
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+DEFAULT_CLOUD_RUN_REGION = "us-central1"
+
+
+class CloudRunJobSteps:
+    """
+    Helper class with Cloud Run job status.
+    Reference: https://cloud.google.com/run/docs/reference/rest/v1/namespaces.jobs#JobStatus
```

Comment:
I highly recommend using the V2 of the Cloud run API, which is much easier to work with (clearer attributes, less Kubernetes-like more like any other GCP API) : https://cloud.google.com/run/docs/reference/rest/v2/projects.locations.jobs#Job

Reply:
I totally agree, but I cannot get the official client to build.

(Putting here the linked issue again https://github.com/apache/airflow/issues/29643)

Reply:
Using the V2 API doesn't mean using the Cloud Run Python client library. 
Here, it seems like you are calling the API via HTTP directly? 
You could hit the /v2/ endpoint instead of the /v1/, it's much easier to use.

This is optional, since you have it working on v1

Reply:
@VinceLegendre can you clarify what is the blocker to use V2?
@lwyszomi is working on updating the google libs in https://github.com/apache/airflow/pull/30067